src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -641,13 +641,19 @@ public override async ValueTask<FormulaValue> Visit(RecordFieldAccessNode node,
             if (left is ErrorValue)
             {
                 return left;
-            }
+            }
+
+            var record = (RecordValue)left;
+
+            if (node.IRContext.IsMutation)
+            {
+                record.ShallowCopyFieldInPlace(node.Field.Value);

```

Comment:
>ShallowCopyFieldInPlace [](http://example.com/codeflow?start=23&length=23)

our default impl of ShallowCopyFieldInPlace throws NotImplemented. 
Could that mean runtime exceptions coming from here?

Reply:
Consider this test: 

Set(x, { t : Table(User)} )
Patch(x.t, First(t), { Name : "...." })

Reply:
see UserInfoTests for tips

Reply:
See notes above about the interface changes.  Specifically added tests for User scenarios such as this one.  In this case, a copy of User will be made in an InMemoryRecord.